compiler/rustc_typeck/src/check/op.rs

```diff
@@ -59,7 +59,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 {
                     // Suppress this error, since we already emitted
                     // a deref suggestion in check_overloaded_binop
-                    err.delay_as_bug();
+                    err.downgrade_to_delayed_bug();
```

Comment:
`delay_as_bug` should emit the diagnostic. Is this a behavior change?

Reply:
There's an emit call already happening when this closure returns -- this closure really should just be mutating the closure